Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -1115,53 +1113,68 @@ namespace xr
             depthInfoView.subImage.imageArrayIndex = 0;
         }
 
-        bool QueryControllerFloatAction(XrAction controllerAction, XrSession session, float& currentFloatState)
+        // Returns true if the action is supported on the current input
+        bool TryUpdateControllerFloatAction(XrAction controllerAction, XrSession session, float& currentFloatState)
         {
             // query input action state
             XrActionStateFloat floatActionState{XR_TYPE_ACTION_STATE_FLOAT};
             XrActionStateGetInfo getInfo{XR_TYPE_ACTION_STATE_GET_INFO};
             getInfo.action = controllerAction;
             XrCheck(xrGetActionStateFloat(session, &getInfo, &floatActionState));
-            auto controllerInfo = sessionImpl.ControllerInfo;
+            if (!floatActionState.isActive)
+            {
+                return false;
+            }
+
             if (floatActionState.changedSinceLastSync)
             {
                 currentFloatState = floatActionState.currentState;
-                return true;
             }
-            return false;
+
+            return true;
         }
 
-       bool QueryControllerBooleanAction(XrAction controllerAction, XrSession session, bool& currentBooleanState)
-       {
-           // query input action state
-           XrActionStateBoolean booleanActionState{XR_TYPE_ACTION_STATE_BOOLEAN};
-           XrActionStateGetInfo getInfo{XR_TYPE_ACTION_STATE_GET_INFO};
-           getInfo.action = controllerAction;
-           XrCheck(xrGetActionStateBoolean(session, &getInfo, &booleanActionState));
-           auto controllerInfo = sessionImpl.ControllerInfo;
-           if (booleanActionState.changedSinceLastSync)
-           {
+        // Returns true if the action is supported on the current input
+        bool TryUpdateControllerBooleanAction(XrAction controllerAction, XrSession session, bool& currentBooleanState)
```

Comment:
I believe you mixed up two things here, action supported and changed since last frame are two different things. you might want probably a different function for action supported. It is a bit ambiguous